### ISSUE TITLE
`brace-style`: `else if` on same line

### ISSUE THREAD

**Issue by Splines** [eslint-stylistic/eslint-stylistic#255]
### Clear and concise description of the problem

Currently the [`brace-style`](https://eslint.style/rules/default/brace-style) does not allow for an option where `else if`, `else` etc. are placed next to the closing `}`. E.g. I'd love to use `1tbs` as style with the following example being correct:

```js
function dummyFunc() {
  const a = "dummy";
  if (a) {
    console.log("test true");
  } else if (a === "dummy") {
    console.log("dummy");
  } else {
    console.log("else");
  }
}
```

### Suggested solution

No suggested solution yet. [This issue](https://github.com/eslint/eslint/issues/4692) might be relevant.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by MetRonnie** [comment]
This seems like a bug to me

**Comment by Splines** [comment]
@MetRonnie
> however leaving it to the default which is supposed to be 1tbs seems to be bugged

With regards to the defaults, see #163 for this, this is not actually a bug but wanted behavior (yes, it might be confusing, but that's up to the maintainers of this project to decide). The [`configuration factory`](https://eslint.style/guide/config-presets#configuration-factory) method is working fine for me and you have fine-grained control over the rules you want to use.

```js
// my eslint.config.mjs file
import js from "@eslint/js";
import stylistic from "@stylistic/eslint-plugin";
import erb from "eslint-plugin-erb";
import globals from "globals";

export default [
  js.configs.recommended,
  // Allow linting of ERB files, see https://github.com/Splines/eslint-plugin-erb
  erb.configs.recommended,
  // Globally ignore the following paths
  {
    ignores: [
      "node_modules/"
    ],
  },
  {
    plugins: {
      "@stylistic": stylistic,
    },
    rules: {
      ...stylistic.configs.customize({
        "indent": 2,
        "jsx": false,
        "quoteProps": "always",
        "semi": "always",
        "braceStyle": "1tbs",
      }).rules,
      "@stylistic/quotes": ["error", "double", { avoidEscape: true }],
      "no-unused-vars": ["warn", { argsIgnorePattern: "^_" }],
    },
    languageOptions: {
      ecmaVersion: 2022,
      sourceType: "module",
      globals: {
        ...customGlobals,
        ...globals.browser,
        ...globals.jquery,
        ...globals.node,
      },
    },
    linterOptions: {
      // see https://github.com/Splines/eslint-plugin-erb/releases/tag/v2.0.1
      reportUnusedDisableDirectives: "off",
    },
  },
];
```
